lib/screens/alerts/alert.ex

```diff
@@ -177,9 +177,13 @@ defmodule Screens.Alerts.Alert do
       |> Enum.flat_map(&format_query_param/1)
       |> Enum.into(%{})
 
-    "alerts"
-    |> Screens.V3Api.get_json(params)
-    |> Screens.Alerts.Parser.parse_result()
+    case Screens.V3Api.get_json("alerts", params) do
+      {:ok, result} ->
+        {:ok, Screens.Alerts.Parser.parse_result(result)}
+
+      _ ->
+        :error
```

Comment:
Previously, a failure to fetch alerts in `Screens.Alerts.Alert.fetch/1` was silently handled by returning `[]`, but that did not match the behavior of our other functions that fetch data from the API—they all explicitly return `{:ok, parsed_data}` or `:error`. Now that we have content like the subway status widget which is always visible, we need to know if alert data is unavailable for some reason, as opposed to there actually being no relevant alerts at the moment. Otherwise, it would incorrectly report the status as "all clear" if alert data is unavailable.

So I updated things to match up and moved the mapping of `:error` -> `[]` (or other values in certain cases) to the calling functions.

Reply:
Oh wait, this literally _is_ `Alert.fetch/1`. It would help if I read the full code context 😑